pkg/controller/daemon/daemon_controller.go

```diff
@@ -842,6 +844,12 @@ func (dsc *DaemonSetsController) podsShouldBeOnNode(
 				// Emit an event so that it's discoverable to users.
 				dsc.eventRecorder.Eventf(ds, v1.EventTypeWarning, FailedDaemonPodReason, msg)
 				podsToDelete = append(podsToDelete, pod.Name)
+			} else if pod.Status.Phase == v1.PodSucceeded {
+				msg := fmt.Sprintf("Found succeeded daemon pod %s/%s on node %s, will try to delete it", pod.Namespace, pod.Name, node.Name)
```

Comment:
I presume these pods might not have a DeletionTimestamp, otherwise this applies https://github.com/kubernetes/kubernetes/issues/116925#issuecomment-1495498973

cc @bobbypage

Reply:
Yes, I shown in here: https://github.com/kubernetes/kubernetes/issues/117018, under Details in the "Additional detailed output:" section. The Succeeded pods don't have deletionTimestamp.

Reply:
Yes, I shown in here: https://github.com/kubernetes/kubernetes/issues/117018, under Details in the "Additional detailed output:" section. The Succeeded pods don't have deletionTimestamp.

Reply:
Yes, graceful node shutdown will not set a deletion timestamp; it only marks the pods as terminal.